ping/ping.c

```diff
@@ -527,7 +536,7 @@ main(int argc, char **argv)
 			rts.opt_so_dontroute = 1;
 			break;
 		case 's':
-			rts.datalen = strtol_or_err(optarg, _("invalid argument"), 0, MAXPACKET - 8);
+			rts.datalen = strtol_or_err(optarg, _("invalid argument"), 0, max_s);
```

Comment:
I guess that I would avoid limiting the upper bound here, because the switch we do later on has a better error message, so I would pass LONG_MAX as the upper bound here instead.

Reply:
datalen is an int. Limitting to LONG_MAX leads to a strange behavior:
```
$ ./builddir/ping/ping -s -1 localhost
./builddir/ping/ping: invalid argument: '-1': out of range: 0 <= value <= 9223372036854775807

$ ./builddir/ping/ping -s 4294967316 localhost
PING localhost (::1) 20 data bytes
28 bytes from localhost (::1): icmp_seq=1 ttl=64 time=0.042 ms
```

Reply:
Ah, missed that, so we should limit it to INT_MAX to get the nicer message for reasonably larger values.

Reply:
FYI I put there `INT_MAX`.

Reply:
@gobenji Thanks for your time!